sqlcollaborative/dbatools
# initial commit of Write-DbaDataTable

with progress bar and sqlbulkcopy options!

How to test
First make a datatable with the new Out-DbaDataTable command
$datatable = Import-Csv C:\temp\customers.csv | Out-DbaDataTable

- Import the table to a non-existing table to an existing database. You should be prompted to use the -AutoCreateTable parameter
- [ ] Write-DbaDataTable -SqlServer sql2014 -InputObject $datatable -Table mydb.dbo.customers

- Next, use the AutoCreateParameter parameter. The import should then work and you should see a progress bar if your csv is big enough
- [ ] Write-DbaDataTable -SqlServer sql2014 -InputObject $datatable -Table mydb.dbo.customers -AutoCreateTable 

- Test to see that this command is sloooowwww and no progress bar is shown. You should see a warning that this way is slow as hell, but it should still proceed
- [ ] $datatable | Write-DbaDataTable -SqlServer sql2014 -Table mydb.dbo.customers

- Attempt to import into a database that doesnt exist
- [ ] $datatable | Write-DbaDataTable -SqlServer sql2014 -Table nonexistingdb.dbo.customers

- Attempt to use it without a fully qualified table name and no -Database
- [ ] $datatable | Write-DbaDataTable -SqlServer sql2014 -Table customers

- Truncate the table prior to import. You should get a prompt
- [ ] Write-DbaDataTable -SqlServer sql2014 -InputObject $datatable -Table mydb.dbo.customers -Truncate

- Do a WhatiF
- [ ] Write-DbaDataTable -SqlServer sql2014 -InputObject $datatable -Table mydb.dbo.customers -Truncate -WhatIf
